src/spriteloader/grf.cpp

```diff
@@ -156,20 +156,20 @@ bool DecodeSingleSprite(SpriteLoader::Sprite *sprite, SpriteFile &file, size_t f
 				}
 
 				for (int x = 0; x < length; x++) {
-					if (colour_fmt & SCC_RGB) {
+					if (colour_fmt.Test(SpriteComponent::RGB)) {
 						data->r = *dest++;
 						data->g = *dest++;
 						data->b = *dest++;
 					}
-					data->a = (colour_fmt & SCC_ALPHA) ? *dest++ : 0xFF;
-					if (colour_fmt & SCC_PAL) {
+					data->a = colour_fmt.Test(SpriteComponent::Alpha) ? *dest++ : 0xFF;
+					if (colour_fmt.Test(SpriteComponent::Palette)) {
 						switch (sprite_type) {
 							case SpriteType::Normal: data->m = file.NeedsPaletteRemap() ? _palmap_w2d[*dest] : *dest; break;
 							case SpriteType::Font:   data->m = std::min<uint8_t>(*dest, 2u); break;
 							default:        data->m = *dest; break;
 						}
 						/* Magic blue. */
-						if (colour_fmt == SCC_PAL && *dest == 0) data->a = 0x00;
+						if (colour_fmt == SpriteComponent::Palette && *dest == 0) data->a = 0x00;
```

Comment:
And here too. Or is this an implied `!colour_fmt.Test(SpriteComponent::Alpha) &&!colour_fmt.Test(SpriteComponent::Rgb)` ?